src/spikeinterface/preprocessing/scale.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import annotations
+
+from spikeinterface.core import BaseRecording
+from spikeinterface.preprocessing.basepreprocessor import BasePreprocessor
+
+
+class ScaleTouVRecording(BasePreprocessor):
+    """
+    Scale raw traces to microvolts (µV).
+
+    This preprocessor uses the channel-specific gain and offset information
+    stored in the recording extractor to convert the raw traces to µV units.
```

Comment:
What happens if the raw data dtype is `int`, is it changed to float? Is it worth mentioning this here?

Reply:
Yes, it does, I think is worth it.

Reply:
If you scale they always have to be float, no matter what right? For neo our scale analog chunk function allows for any dtype with a default of float.

Reply:
Yes I think I am generally misunderstanding how the scaling works, I will ask more clarifying questions 😅 (thanks for your patience, I promise this will be useful as will write a docs article :D) :

Are the gains and offsets always to uV? (rather than say V). Like I thought it could be possible that you have an integer recording and the associated gains and offsets are to V, so when you do `return_scaled=True` you get the recording in V. Then one thing this might do is change the recording such that it is still integer but when `return_scaled=True` it returns as `uV` rather than `V`.

Instead, (I believe) the purpose of this to change the raw datatype such that the integer data is set to float, and scaled by the scaling factors, and then the gain and offset are changed to 1 and 0 forever more? and `return_scaled=True` and `return_scaled=False` have the same behaviour?

If the original recording is a float, but also has some gains and offsets such that `return_scaled=False` and `return_scaled=True` give different output (the latter also float but scaled to `uV`), this will simply scale the raw data to `uV` also? (and set gain 1 offset 0).

Reply:
Yes, by design we write gains and scales so when you multiply by the scale and add the offset you end up with units in uV. Always uV.

No, this preprocessor should not alter the gains and offsets of the preprocessed recording. It just does the scaling at some point of the chain so you can be sure that anything you do either will be in "natural" units in opposition to "storage" units.

Reply:
I think this is in our documents somewhere, but this would be another great point to add to a glossary. Anything scaled will be in uV. It's a choice the library has made so any other units have to be accomplished by the end-user.

Reply:
Yes!